app/controllers/svg_sprite_controller.rb

```diff
@@ -49,4 +49,25 @@ def icon_picker_search
       render json: icons.take(200), root: false
     end
   end
+
+  def svg_icon
+    no_cookies
+
+    RailsMultisite::ConnectionManagement.with_hostname(params[:hostname]) do
+      params.permit(:color)
+      name = params.require(:name)
+      icon = SvgSprite.search(name)
+
+      if icon.blank?
+        render body: nil, status: 404
+      else
+        doc = Nokogiri.XML(icon)
+        doc.at_xpath("symbol").name = "svg"
+        doc.at_xpath("svg")['xmlns'] = "http://www.w3.org/2000/svg"
+        doc.at_xpath("svg")['fill'] = "##{params[:color]}" if params[:color]
+
+        render plain: doc, disposition: nil, content_type: 'image/svg+xml'
+      end
```

Comment:
Maybe add some cache headers? So the CDN can cache it for us. Since the icons can change between updates, I'd say a day?